File: JavaCodingQuestions/bubbleSort.java
import java.util.*;
class Main {
    static void bubbleSort(int array[])
    {
        int size = array.length;

        for(int i = 0; i < size; i++) {
            for(int j =0; j < size; j++) {
                if(array[j] > array[j+1]) {
                    int temp = array[j];
                    array[j] = array[j+1];
                    array[j+1] = temp;
                }
            }
        }
    }

    public static void main(String args[]) {
        int[] data = {4,65,5,76,34,23};

        Main.bubbleSort(data);

        System.out.println("Sorted Array: ");
        System.out.println(Arrays.toString(data));
    }
}

File: CppCodingQuestions/bubbleSort.cpp
#include<iostream>
using namespace std;

void bubbleSort(int array[], int size) {
    for(int step = 0; step < size; ++step) {
        for(int i = 0; i<size - step; ++i) {
            if(array[i] > array[i+1]) {
                int temp = array[i];
                array[i] = array[i + 1];
                array[i+1] = temp;
            }
        }
    }
}
void printArray(int array[], int size) {
    for(int i = 0; i < size; ++i) {
        cout<< "  "<<array[i];
    }
        cout<<"/n";
}
int main() {
    int data = {32,63,12,56,224,67,23};
    bubbleSort(data,size);
    cout<<"Sorted Array : ";
    printArray(data,size);
}

